node-src/lib/tasks.ts

```diff
@@ -1,15 +1,29 @@
 import chalk from 'chalk';
 import Listr from 'listr';
 import pluralize from 'pluralize';
-import { Context, Task } from '../types';
+import { Context, Task, TaskName } from '../types';
 
 type ValueFn = string | ((ctx: Context, task: Task) => string);
 
-export const createTask = ({ title, steps, ...config }): Listr.ListrTask<Context> => ({
+type TaskInput = Omit<Listr.ListrTask<Context>, 'task'> & {
+  name: TaskName;
+  steps: ((ctx: Context, task: Listr.ListrTaskWrapper<Context> | Task) => void | Promise<void>)[];
+};
+
+export const createTask = ({
+  name,
+  title,
+  steps,
+  ...config
+}: TaskInput): Listr.ListrTask<Context> => ({
   title,
   task: async (ctx: Context, task: Listr.ListrTaskWrapper<Context>) => {
+    ctx.task = name;
```

Comment:
I don't like exposing task names as an API. People will start to take the order of tasks for granted as part of the API, but I want to have the flexibility to change the order of tasks, or even add/remove tasks as we see fit.

Reply:
Have discussed and decided this API is basically internal for now. We will mark it as such (`experimental_`) and not worry too much about it.